### PR TITLE
feat(node): support `username` and `_password` in `.npmrc` file

### LINKED ISSUE #23950: .npmrc: support `username` and `_password` auth configuration
https://docs.npmjs.com/cli/v10/configuring-npm/npmrc#auth-related-configuration

Needs to be addressed after https://github.com/denoland/deno/pull/23560 lands.
Ref https://github.com/denoland/deno/issues/16105

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #23950

### CODE REVIEW

**Comment by bartlomieju** [comment]
> Erroring when only username or _password is defined can be handled in a follow-up PR.

I think we should do it in this PR

**Review by bartlomieju — COMMENTED** [review]
Actually we should verify what `npm` CLI does in such case.

**Comment by iuioiua** [comment]
npm throws when `username` or `_password` isn't provided without the other ([ref](https://github.com/npm/cli/blob/4e81a6a4106e4e125b0eefda042b75cfae0a5f23/workspaces/config/test/index.js#L846)). I've added this behavior in now.

File: cli/npm/common.rs [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
+    )));
   }
 
-  None
+  let (username, password) = (
+    registry_config.username.as_ref(),
+    registry_config.password.as_ref(),
+  );
+  if (username.is_some() && password.is_none())
+    || (username.is_none() && password.is_some())
```
Comment (bartlomieju) [inline cli/npm/common.rs:33]: Nitpick, but you can probably compat this and the next conditional by using a match like so:
```rust
match (username, password) {
  (Some(username), Some(password)) => {...},
  (Some(_) , None_ => {...},
  (None, Some(_) => {...},
  _ => {}
}
```

**Review by bartlomieju — APPROVED** [review]
LGTM, nice work!

File: cli/npm/managed/cache/registry_info.rs [hunk 2]
```diff
@@ -191,11 +191,12 @@ impl RegistryInfoDownloader {
   fn create_load_future(self: &Arc<Self>, name: &str) -> LoadFuture {
     let downloader = self.clone();
     let package_url = self.get_package_url(name);
-    let registry_config = self.npmrc.get_registry_config(name);
-    let maybe_auth_header = maybe_auth_header_for_npm_registry(registry_config);
+    let registry_config = self.npmrc.get_registry_config(name).clone();
```
Comment (dsherret) [inline cli/npm/managed/cache/registry_info.rs:194]: A clone could be avoided here by instead doing:

```rs
let maybe_auth_header = match maybe_auth_header_for_npm_registry(&registry_config) {
  Ok(maybe_auth_header) => maybe_auth_header,
  Err(err) => return std::future::ready(Err(Arc::new(err))).boxed_local(),
};
```

**Review by dsherret — APPROVED** [review]
LGTM too.

File: cli/npm/common.rs (hunk 1 above)
Comment (iuioiua) [reply cli/npm/common.rs:33]: I'll revisit this once when we add the `certfile` and `keyfile` fields.

**Pull request merged**